### ISSUE TITLE
String parsing/escaping is different between 219 and 222

### ISSUE THREAD

**Issue by utezduyar** [systemd/systemd#701]
Following service was working fine with systemd 219 and now it complains that -N argument needs a parameter. I think systemd swallows `''`.

I can workaround this by changing the service to following but I think it is not the right way.
ExecStart=/bin/sh -c "/usr/bin/ssh-keygen -t rsa -f /etc/ssh/ssh_host_rsa_key -N \'\' -b 2048"

```
cat /usr/lib/systemd/system/sshdgenkeys.service 
[Unit]
Description=SSH Key Generation
ConditionPathExists=|!/etc/ssh/ssh_host_rsa_key
ConditionPathExists=|!/etc/ssh/ssh_host_rsa_key.pub

[Service]
Type=oneshot
RemainAfterExit=yes
ExecStart=/usr/bin/ssh-keygen -t rsa -f /etc/ssh/ssh_host_rsa_key -N '' -b 2048


[242688.022425] systemd[1]: Starting SSH Key Generation...
[242688.052557] ssh-keygen[2946]: option requires an argument -- N
[242688.057286] ssh-keygen[2946]: usage: ssh-keygen [options]
[242688.061847] ssh-keygen[2946]: Options:
[242688.066124] ssh-keygen[2946]: -A          Generate non-existent host keys for all key types.
[242688.069386] ssh-keygen[2946]: -a number   Number of KDF rounds for new key format or moduli primality tests.
[242688.072520] ssh-keygen[2946]: -B          Show bubblebabble digest of key file.
[242688.075385] ssh-keygen[2946]: -b bits     Number of bits in the key to create.
[242688.078254] ssh-keygen[2946]: -C comment  Provide new comment.
[242688.083343] ssh-keygen[2946]: -c          Change comment in private and public key files.
[242688.088529] ssh-keygen[2946]: -D pkcs11   Download public key from pkcs11 token.
[242688.092967] ssh-keygen[2946]: -e          Export OpenSSH to foreign format key file.
[242688.095836] ssh-keygen[2946]: -F hostname Find hostname in known hosts file.
[242688.098678] ssh-keygen[2946]: -f filename Filename of the key file.
[242688.101733] ssh-keygen[2946]: -G file     Generate candidates for DH-GEX moduli.
[242688.104612] ssh-keygen[2946]: -g          Use generic DNS resource record format.
[242688.107528] ssh-keygen[2946]: -H          Hash names in known_hosts file.
[242688.110785] ssh-keygen[2946]: -h          Generate host certificate instead of a user certificate.
[242688.114033] ssh-keygen[2946]: -I key_id   Key identifier to include in certificate.
[242688.117276] ssh-keygen[2946]: -i          Import foreign format to OpenSSH key file.
[242688.121435] ssh-keygen[2946]: -J number   Screen this number of moduli lines.
[242688.126870] ssh-keygen[2946]: -j number   Start screening moduli at specified line.
[242688.132516] ssh-keygen[2946]: -K checkpt  Write checkpoints to this file.
[242688.135698] ssh-keygen[2946]: -k          Generate a KRL file.
[242688.139085] ssh-keygen[2946]: -L          Print the contents of a certificate.
[242688.144431] ssh-keygen[2946]: -l          Show fingerprint of key file.
[242688.148662] ssh-keygen[2946]: -M memory   Amount of memory (MB) to use for generating DH-GEX moduli.
[242688.152967] ssh-keygen[2946]: -m key_fmt  Conversion format for -e/-i (PEM|PKCS8|RFC4716).
[242688.157143] ssh-keygen[2946]: -N phrase   Provide new passphrase.
[242688.162945] ssh-keygen[2946]: -n name,... User/host principal names to include in certificate
[242688.169781] ssh-keygen[2946]: -O option   Specify a certificate option.
[242688.174815] ssh-keygen[2946]: -o          Enforce new private key format.
[242688.178487] ssh-keygen[2946]: -P phrase   Provide old passphrase.
[242688.182699] ssh-keygen[2946]: -p          Change passphrase of private key file.
[242688.186533] ssh-keygen[2946]: -Q          Test whether key(s) are revoked in KRL.
[242688.190449] ssh-keygen[2946]: -q          Quiet.
[242688.194483] ssh-keygen[2946]: -R hostname Remove host from known_hosts file.
[242688.198291] ssh-keygen[2946]: -r hostname Print DNS resource record.
[242688.204114] ssh-keygen[2946]: -S start    Start point (hex) for generating DH-GEX moduli.
[242688.211668] ssh-keygen[2946]: -s ca_key   Certify keys with CA key.
[242688.215970] ssh-keygen[2946]: -T file     Screen candidates for DH-GEX moduli.
[242688.220613] ssh-keygen[2946]: -t type     Specify type of key to create.
[242688.224771] ssh-keygen[2946]: -u          Update KRL rather than creating a new one.
[242688.229048] ssh-keygen[2946]: -V from:to  Specify certificate validity interval.
[242688.233411] ssh-keygen[2946]: -v          Verbose.
[242688.237516] ssh-keygen[2946]: -W gen      Generator to use for generating DH-GEX moduli.
[242688.244856] ssh-keygen[2946]: -y          Read private key file and print public key.
[242688.252194] ssh-keygen[2946]: -Z cipher   Specify a cipher for new private key format.
[242688.256296] ssh-keygen[2946]: -z serial   Specify a serial number.
[242688.354792] systemd[1]: sshdgenkeys.service: Main process exited, code=exited, status=1/FAILURE
[242688.365014] systemd[1]: Failed to start SSH Key Generation.
[242688.373984] systemd[1]: sshdgenkeys.service: Unit entered failed state.
[242688.377753] systemd[1]: sshdgenkeys.service: Failed with result 'exit-code'.
```

**Comment by richardmaw-codethink** [comment]
I think pull request #44 may have caused this change.
From my reading of how unquote_first_word works, it doesn't produce an empty result when entering the quoted section, and there's nothing in it to add to this result.

**Comment by zonque** [comment]
cc @filbranden

**Comment by richardmaw-codethink** [comment]
@utezduyar does #704 solve your problem?

**Comment by utezduyar** [comment]
Thanks! It worked for me.

**Comment by mbiebl** [comment]
Looks like something which should be fixed for v223

**Comment by zonque** [comment]
Fixed in #704
